bundles/com.espressif.idf.core/src/com/espressif/idf/core/bug/BugReportGenerator.java

```diff
@@ -0,0 +1,230 @@
+/*******************************************************************************
+ * Copyright 2025 Espressif Systems (Shanghai) PTE LTD.
+ * All rights reserved. Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.core.bug;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.Platform;
+
+import com.espressif.idf.core.logging.Logger;
+import com.espressif.idf.core.util.FileUtil;
+import com.espressif.idf.core.util.StringUtil;
+
+/**
+ * This class generates a bug report zip file containing:
+ * 1. Installed tools information
+ * 2. Product information
+ * 3. Basic system information (OS, Arch, Memory)
+ * 4. IDE metadata logs
+ * 5. eim logs (if available)
+ * 
+ * The generated zip file is named with a timestamp and stored in the workspace directory.
+ * @author Ali Azam Rana
+ *
+ */
+public class BugReportGenerator
+{
+	private static final String bugReportDirectoryPrefix = "bug_report_"; //$NON-NLS-1$
```

Comment:
nit: the naming convention for a final static field is all uppercase letters with underscores separating words.